godtools/App/Share/SwiftUI Views/SearchBar/SearchBar.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  SearchBar.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 8/29/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import SwiftUI
+
+// adapted from https://www.appcoda.com/swiftui-search-bar/
+
+struct SearchBar: View {
+    
+    @Binding var searchText: String
+    let localizationServices: LocalizationServices
+    
+    @State private var isEditing = false
+        
+    var body: some View {
+        HStack {
+         
+            TextField("", text: $searchText, onEditingChanged: { _ in
+                self.isEditing = true
+            }, onCommit: {
+                DispatchQueue.main.async {
+                    
+                    self.isEditing = false
+                }
+            })
+            .padding(7)
+            .padding(.horizontal, 27)
+            .background(Color.white)
+            .cornerRadius(6)
+            .overlay(
+                HStack {
+                    
+                    Image(systemName: "magnifyingglass")
+                        .foregroundColor(.gray)
+                        .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
+                        .padding(.leading, 10)
+                    
+                    if isEditing {
+                        Button(action: {
+                            self.searchText = ""
+                            
+                        }) {
+                            Image(systemName: "multiply.circle.fill")
+                                .foregroundColor(.gray)
+                                .padding(.trailing, 10)
+                        }
+                    }
+                }
+            )
+            
+            if isEditing {
+                Button(action: {
+                    DispatchQueue.main.async {
```

Comment:
It seemed strange to me that DispatchQueue.main.async is needed here, but it also seems that the state won't trigger without it?  Is that what you found?

Reply:
I *think* it's something about `resignFirstResponder`.  When I remove that UIApplication call, the `cancel` button at least works properly (but the keyboard doesn't get dismissed).  I can look into this a little more, maybe there's a different way this is supposed to happen in SwiftUI.